src/vanilla/utils/freezeAtom.ts

```diff
@@ -2,15 +2,17 @@ import type { Atom, WritableAtom } from '../../vanilla.ts'
 
 const frozenAtoms = new WeakSet<Atom<any>>()
 
-const deepFreeze = (obj: unknown) => {
-  if (typeof obj !== 'object' || obj === null) return
-  Object.freeze(obj)
-  const propNames = Object.getOwnPropertyNames(obj)
+const deepFreeze = (value: unknown) => {
```

Comment:
This should reveal the bug.

```suggestion
const deepFreeze = <T>(value: T): T => {
```